File: 算法/整数拆分.php
<?php

class Solution {

    /**
     * @param Integer $n
     * @return Integer
     */
    function integerBreak($n) {
        if($n < 4) {
            return $n - 1;
        }

        $res = 1;
        while($n > 4) {
            $res *= 3;
            $n -= 3;
        }

        return $n == 0 ? $res : $res * $n;
    }
}

$solution = new Solution();
echo $solution->integerBreak(4);

File: 算法/整数拆分.go
package 算法

func integerBreak(n int) int {
	if n < 4 {
		return n - 1
	}

	res := 1
	for n > 4 {
		n -= 3
		res *= 3
	}

	if n == 0 {
		return res
	}
	return res * n
}
